src/libmongoc/src/mongoc/mongoc-ocsp-cache.c

```diff
@@ -80,6 +80,26 @@ update_entry (cache_entry_list_t *entry,
    entry->cert_status = cert_status;
    entry->reason = reason;
 }
+
+#if OPENSSL_VERSION_NUMBER >= 0x10101000L
+static int
+_cmp_time (ASN1_TIME *a, ASN1_TIME *b)
+{
+   return ASN1_TIME_compare (a, b);
+}
+#else
+static int
+_cmp_time (ASN1_TIME *a, ASN1_TIME *b)
+{
+   /* For older OpenSSL, always report that "a" is before "b". I.e. do not
+    * replace the entry.
+    * If a driver would accept a stapled OCSP response and that response has a
+    * later nextUpdate than the response already in the cache, drivers SHOULD
+    * replace the older entry in the cache with the fresher response. */
+   return -1;
```

Comment:
I'll admit, this is not ideal. After searching a bit, and checking with Shreyas to see how the server polyfilled the time comparison, it seems like we'd need to parse the string representation of the time to something we can easily compare:
https://github.com/mongodb/mongo/blob/f627ff829ba96487fccb6945d0c2807b1bb51d52/src/mongo/util/net/ssl_manager_openssl.cpp#L375-L412

It doesn't look like it's required by the spec to check `nextUpdate` when updating the cache. By always comparing "a" to be before "b", we won't overwrite cache entries. They'll still get removed when they're detected as being expired by `OCSP_check_validity`.

If this seems ok with you, I'll create a ticket to do this as post-4.4 follow-up.

Reply:
Created https://jira.mongodb.org/browse/CDRIVER-3707